### PR TITLE
Story/tp65735 auth

### PR DESCRIPTION
**Pull request by bakerac4**

### CODE REVIEW

File: pep/app/authenticators/pep.ts
Comment (billdami) [inline pep/app/authenticators/pep.ts:27]: you prob just didnt get to this yet, but we'll want to make this base url for the auth endpoints a new env var, like `AUTH_BASE_URL` and then import it here from `ENV`. to add new env vars you need to update a couple files:

- add it to every `.env-*` file in the root (leaving the value blank for staging/prod)
- pep/config/dotenv.js
- pep/app/config/environment.d.ts
- pep/config/environment.js

File: .env-production
Comment (billdami) [inline .env-production:9]: @lhorne-gavant just making sure you're aware of this new env var we'll want to add to our build scripts

File: pep/app/services/ajax.ts
Comment (billdami) [inline pep/app/services/ajax.ts:59]: just noting that i think we will ultimately need this left in (as alan mentioned the apis will want to be sending/sharing cookies for some other stuff), so if if its causing issue with PaDs, we may need to have him address that. For the time being, i'd probably just comment this out instead of deleting it.

File: pep/app/authenticators/pep.ts
Comment (billdami) [inline pep/app/authenticators/pep.ts:35]: not sure if @adistasio talked to you about this yet, but apparently we are never doing OAuth now, so we'll need some to add some logic to the authenticator that automatically invalidates the local session when the returned session expires.

I'll most likely look somewhat similar to the `_scheduleAccessTokenRefresh` logic we have in our oauth-based authenticators (see CC or JRE), but instead of refreshing, itll just call `invalidate()` once the expire date is hit. This timer would most likely need to be kicked off both in `authenticate()` and `restore()`.

If we want to do this in a follow up PR, thats probably fine too.

File: pep/app/pods/application/adapter.ts
Comment (billdami) [inline pep/app/pods/application/adapter.ts:34]: this might be better off living in `types/pep/api.d.ts`

File: pep/app/pods/application/adapter.ts
Comment (billdami) [inline pep/app/pods/application/adapter.ts:82]: we've custom defined this `authenticated ` session data interface ourselves, in `types/ember-simple-auth/services/session.d.ts`. So its probably better to just change it there directly, to `PepSecureAuthenticatedData` so we don't need to convert here or anywhere else.

File: pep/app/services/ajax.ts
Comment (billdami) [inline pep/app/services/ajax.ts:30]: same here as the last comment, if we just adjust our already custom session data interface, we won't need to convert it everywhere.

**Review by billdami — CHANGES_REQUESTED** [review]

**Comment by billdami** [comment]
@bakerac4 does this fix #27?

File: pep/app/pods/application/adapter.ts
Comment (bakerac4) [reply pep/app/pods/application/adapter.ts:82]: I'd tend to disagree with this on the basis that eventually we won't be specifying those types as they will provide them and in that case we will have to do this anyways

Comment (billdami) [reply pep/app/pods/application/adapter.ts:73]: @bakerac4 totally agree on that point (that we can remove all ember-simple-auth specific types) once they have types defined themselves. 

But in this case, our `authenticated` interface is custom and API-dependent (the existing `SessionAuthenticatedData` interface we defined is for OAuth2, so wouldn't even apply here even if ESA had types for oauth2 structures, since we made our own custom authenticator). So i still think that if/when we get types from ESA, we would and should be still be directly extending/typing that one interface at the module level to be what really is, instead of it "starting" as oauth2.

File: pep/app/services/pep-session.ts
Comment (billdami) [inline pep/app/services/pep-session.ts:27]: we should rename this class to `PepSessionService` and import it by the same name elsewhere, just to be consistent with all the other service's "Session" suffix and matching the filename to the class name.

I can do this in my branch tho once I merge this into mine, if you want. as I already will have to fix a few conflicts I see.

**Pull request merged**
